### DISCUSSION TITLE
Regarding the use of cudaMemcpyAsync and cudaStreamSynchronize, the data is not copied from the device back to the host.

### DISCUSSION THREAD

**Discussion by HCSR-Lab** [accel-sim/accel-sim-framework#183] [Q&A]
**Problem**: When I use cudaMemcpyAsync, the data is not copied from the device back to the host.

Should I, or roughly where, modify the GPGPU-Sim source code?

**Comment by JRPan** [comment]
It should be supported. Something else is probably wrong.
